### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by jrit**
@vokal/web roadhouse has returned (as a repo) as a set of directives. Additional tests and documentation will be forthcoming. It is possible this might be more useful at some point with the directives broken out into their own repos, but that isn't needed for my purposes at the moment.

### CODE REVIEW

File: tests/utils.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        "var scope = angular.element( document.body ).injector().get( '$rootScope' );"
+        + "scope['" + key + "'] = JSON.parse( '" + JSON.stringify( val ) + "' );"
+        + "scope.$apply();" );
+};
+
+utils.applyScope = function ()
+{
+    browser.executeScript(
+        "var scope = angular.element( document.body ).injector().get( '$rootScope' );"
+        + "scope.$apply();" );
```
Comment (dmaloneycalu) [inline tests/utils.js]: This doesn't need to be saved to a variable, can be run directly as `"angular.element( document.body ).injector().get( '$rootScope' ).$apply();"`

File: package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    "grunt-browserify": "^4.0.1",
+    "grunt-contrib-clean": "^0.6.0",
+    "grunt-contrib-less": "^1.1.0",
+    "grunt-istanbul": "^0.6.1",
+    "grunt-protractor-coverage": "^0.2.15",
+    "protractor": "^2.5.1"
+  },
+  "dependencies": {
+    "angular-datetime": "github:vokal/angular-datetime",
+    "angular-mocks": "github:vokal/bower-angular-mocks#52745f56ed",
```
Comment (StoicJester) [inline package.json]: why is mocks both a dev and normal dependency?

Comment (jrit) [inline package.json]: it is a mistake, should just be dev

File: tests/utils.js (hunk 1 above)
Comment (jrit) [inline tests/utils.js]: really more of a consistency thing with the other 2 here, but I suppose it should be removed anyway

File: test/rhTable.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        expect( $$( ".rh-table tbody tr" ).count() ).toBe( 2 ); // 2 body rows
+        expect( $$( ".rh-table tbody tr:first-child .glyphicon-edit" ).count() ).toBe( 1 ); // edit button
+        expect( $$( ".table-controls" ).count() ).toBe( 1 ); // add button visible
+    } );
+
+    it( "should have non-view columns", function ()
+    {
+        utils.set( "definition", { meta: {}, id: { canViewList: false }, name: {} } );
+
+        expect( $$( ".rh-table thead th" ).count() ).toBe( 2 ); // 2 columns
```
Comment (dmaloneycalu) [inline test/rhTable.js]: Add a check prior to setting the definition to show this action changes the number of columns.

Comment (Tathanen) [inline test/rhTable.js]: :+1:

File: test/rhTable.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        expect( $$( ".rh-table tbody tr.deleted" ).count() ).toBe( 0 ); // no deleted rows
+        expect( $( ".ngdialog .delete-confirm" ).isPresent() ).toBe( false );
+
+        $$( ".rh-table tbody tr .glyphicon-edit" ).get( 2 ).click();
+        $( ".ngdialog .initial" ).element( by.buttonText( "Delete" ) ).click();
+        $( ".ngdialog .delete-confirm" ).element( by.buttonText( "Yes, Delete" ) ).click();
+        browser.sleep( 1000 );
+
+        utils.expectNgDialogIsHidden();
+        expect( $$( ".rh-table tbody tr.deleted" ).count() ).toBe( 1 ); // 1 deleted row
```
Comment (dmaloneycalu) [inline test/rhTable.js]: Interesting decision to not remove a deleted item from the list.

**Comment by dmaloneycalu** [comment]
One test addition, otherwise it LGTM. Lot of stuff going on here and will have to see it working in action.

File: test/rhTable.js (hunk 4 above)
Comment (jrit) [inline test/rhTable.js]: I found it to be somewhat disorienting to just have it be gone because I'd forget which row I was on

**Pull request merged**